qiime/sdk/tests/test_visualization.py

```diff
@@ -240,6 +241,63 @@ def test_load_from_externally_created_zipfile(self):
             }
             self.assertEqual(fps, expected)
 
+    def test_extract(self):
+        fp = os.path.join(self.test_dir.name, 'visualization.qzv')
+        visualization = Visualization._from_data_dir(self.data_dir,
+                                                     self.provenance)
+        visualization.save(fp)
+
+        Visualization.extract(fp)
+
+        contents = [
+            'visualization/VERSION',
+            'visualization/metadata.yaml',
+            'visualization/README.md',
+            'visualization/data/index.html',
+            'visualization/data/css/style.css']
+
+        for fp in contents:
+            expected_fp = os.path.join(tempfile.gettempdir(), fp)
+            self.assertTrue(os.path.exists(expected_fp),
+                            'File %s was not extracted.' % fp)
+
+    def test_extract_alt_output_dir(self):
+        fp = os.path.join(self.test_dir.name, 'visualization.qzv')
+        visualization = Visualization._from_data_dir(self.data_dir,
+                                                     self.provenance)
+        visualization.save(fp)
+
+        output_dir = os.path.join(self.test_dir.name, 'viz-extract-test')
+        Visualization.extract(fp, output_dir=output_dir)
+
+        contents = [
+            'visualization/VERSION',
+            'visualization/metadata.yaml',
+            'visualization/README.md',
+            'visualization/data/index.html',
+            'visualization/data/css/style.css']
+
+        for fp in contents:
+            expected_fp = os.path.join(output_dir, fp)
+            self.assertTrue(os.path.exists(expected_fp),
+                            'File %s was not extracted.' % fp)
+
+    def test_get_indices_single(self):
+        visualization = Visualization._from_data_dir(self.data_dir,
+                                                     self.provenance)
+        actual = visualization.get_indices()
+        expected = {'data/index.html'}
+        self.assertEqual(actual, expected)
+
+    def test_get_indices_multiple(self):
+        most_common_viz(self.data_dir,
+                        collections.Counter(range(42)))
+        visualization = Visualization._from_data_dir(self.data_dir,
```

Comment:
This particular path is causing `_data_dir` to exist on the `Archiver`, but I don't think that is generally true. @jairideout can you confirm?